Circle.cpp

```diff
@@ -1,14 +1,37 @@
 #ifndef Circle_cpp
 #define Circle_cpp
+
+#define _USE_MATH_DEFINES
+
 #include <iostream>
-#include "Draw.cpp"
+#include <math>
+
+#include "Shape.cpp"
+#include "Area.cpp"
+#include "Input.cpp"
+
 using namespace std;
-class Circle : public Draw
+class Circle : public Shape , public Input
 {
 public:
     void drawShape() override
     {
-        cout << "circle" << endl;
+        cout << "circle" << endl; 
+    }
+    int inputRadius() override 
+    {
+        int radius;
+        cout << "Enter Circle Radius" << endl;
+        cin >> radius;
+        return radius;
+    }
+    int CalculateArea() override
+    {
+        int radius = inputRadius ();
```

Comment:
We should not be calling inputRadius function from the `calculateArea` function. Whether the work/statements are directly written or called behind a function (in this case `inputRadius`).

A function's job is to perform one action and one action alone. You can call a function from inside of a function but the outer function name's meaning should always encapsulate everything that is going on inside.

I suggest to call the input function from the main function. before calculating the area. Which in our case is acting as the driver.